invoke-ai/InvokeAI
# ui: redesign followups 8

## What type of PR is this? (check all applicable)



## Have you discussed this change with the InvokeAI team?

      
## Have you updated all relevant documentation?


## Description

Minor fixes/enhancements:
- disable cancel queue hotkeys in input elements (prevented you from typing uppercase X)
- default scale mode is now auto, closes  #5439
- fix embedding popover not capturing focus while loading TIs, closes  #5442
- enable control adapter autoconfig when switching the ca model, closes #5440
- use JSON.stringify for `roarr` error message serializer (fixes new issue w/ truncated contexts in console)
- add $socket to window when in dev mode to assist in debugging
- simplify redux actions related to sockets, may save us a few microseconds but more importantly its much cleaner
- add debuggingLoggerMiddleware. similar to redux devtools, but devtools will only keep the latest however many actions. browser console has much higher history. this must be manually added to middleware to enable.
- bump deps
- add error handling for `redux-remember` persist/rehydrate actions (better visibliity when there is a problem)
- add migrations for persisted redux state, [fixes issue with UI crash after app update](https://discord.com/channels/1020123559063990373/1149506274971631688/1193203811402121237). This is a bare-bones implementation with no type safety within the migration logic. In the future we might want to model app state w/ a schema/zod to get better safety. There *is*, however, good type safety surrounding the migration logic, ensuring that persisted slices are set up properly.

## Related Tickets & Documents


- Related https://discord.com/channels/1020123559063990373/1149506274971631688/1193203811402121237
- Closes #5439
- Closes #5444 
- Closes #5440

## QA Instructions, Screenshots, Recordings

The most important thing to test is that upgrades do not crash the UI. Note that this fix only works for v3.5.1 to v3.6.0 and onward. This will not fix other occurrences of the issue when updating from earlier version.

To reproduce the issue:
- Check out tag v3.5.1
- Run UI in dev mode
- Reset web UI
- Kill dev server
- Check out `main`
- Run UI in dev mode
- Refresh browser

You should get an error: `TypeError: Cannot read properties of null (reading 'id')`

Then, to test this PR:
- Reset web UI
- Redo the steps above, except instead of `main`, check out this PR branch

You should get no error and the app loads fine.



## Merge Plan

This PR can be merged when approved. Please reproduce the crash and fix as described above before approving.
